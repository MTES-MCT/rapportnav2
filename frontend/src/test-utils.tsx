import React, { ReactElement } from 'react'
import { render, renderHook, RenderOptions } from '@testing-library/react'
import UIThemeWrapper from './ui/ui-theme-wrapper'
import { BrowserRouter } from 'react-router-dom'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <UIThemeWrapper>
      <BrowserRouter>{children}</BrowserRouter>
    </UIThemeWrapper>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options })

const customRenderHook = (hook: any, options?: Omit<RenderOptions, 'wrapper'>) =>
  renderHook(() => hook(), { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
export { customRenderHook as renderHook }
