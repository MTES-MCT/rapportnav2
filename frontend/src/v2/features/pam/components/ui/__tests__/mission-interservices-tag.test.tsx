import { render, screen } from '../../../../../../test-utils.tsx'
import { describe, it, expect } from 'vitest'
import MissionInterServicesTag from '../mission-interservices-tag'

describe('MissionInterServicesTag', () => {
  it('renders the tag text', () => {
    render(<MissionInterServicesTag />)
    expect(screen.getByText('Inter-services')).toBeInTheDocument()
  })

  it('matches snapshot', () => {
    const { container } = render(<MissionInterServicesTag />)
    expect(container).toMatchSnapshot()
  })
})
