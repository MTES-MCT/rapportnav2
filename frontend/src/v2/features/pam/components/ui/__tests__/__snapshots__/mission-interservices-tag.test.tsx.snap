// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MissionInterServicesTag > matches snapshot 1`] = `
<div>
  <span
    class="sc-fOOuSg iMvLlc Element-Tag"
    style="height: auto;"
  >
    <p
      class="sc-iRcyzz sc-fFmWCW iyEmXO hAFmCO"
      color="#707785"
      font-size="13"
      weight="medium"
    >
      Inter-services
    </p>
  </span>
</div>
`;
